File: app/src/main/java/com/jingxi/smartlife/pad/mvp/login/model/LoginModel.java
package com.jingxi.smartlife.pad.mvp.login.model;

import com.jingxi.smartlife.pad.mvp.MainAppApiService;
import com.jingxi.smartlife.pad.mvp.login.contract.LoginContract;
import com.wujia.businesslib.base.BaseModel;
import com.wujia.businesslib.data.LoginDTO;
import com.wujia.businesslib.data.RootResponse;
import com.wujia.lib_common.data.network.RxUtil;
import com.wujia.lib_common.utils.SystemUtil;

import io.reactivex.Flowable;

/**
 * author ：shenbingkai@163.com
 * date ：2019-01-27
 * description ：
 */
public class LoginModel extends BaseModel implements LoginContract.Model {



    @Override
    public Flowable<LoginDTO> login(String mobile, String captcha, String padSn) {
        return mHttpHelper.create(MainAppApiService.class).login(mobile, captcha, padSn).compose(RxUtil.<LoginDTO>rxSchedulerHelper());
    }

    @Override
    public Flowable<RootResponse> getCode(String mobile) {
        return mHttpHelper.create(MainAppApiService.class).getCode(mobile, SystemUtil.getSerialNum()).compose(RxUtil.<RootResponse>rxSchedulerHelper());
    }


}

File: app/src/main/java/com/jingxi/smartlife/pad/mvp/login/model/LoginModel.kt
package com.jingxi.smartlife.pad.mvp.login.model

import com.jingxi.smartlife.pad.mvp.MainAppApiService
import com.jingxi.smartlife.pad.mvp.login.contract.LoginContract
import com.wujia.businesslib.base.BaseModel
import com.wujia.businesslib.data.LoginDTO
import com.wujia.lib_common.base.RootResponse
import com.wujia.lib_common.data.network.RxUtil
import com.wujia.lib_common.utils.SystemUtil

import io.reactivex.Flowable

/**
 * author ：shenbingkai@163.com
 * date ：2019-01-27
 * description ：
 */
class LoginModel : BaseModel(), LoginContract.Model {


    override fun login(mobile: String, captcha: String, padSn: String): Flowable<LoginDTO> {
        return mHttpHelper.create(MainAppApiService::class.java)!!.login(mobile, captcha, padSn).compose(RxUtil.rxSchedulerHelper())
    }

    override fun getCode(mobile: String): Flowable<RootResponse> {
        return mHttpHelper.create(MainAppApiService::class.java)!!.getCode(mobile, SystemUtil.getSerialNum()!!).compose(RxUtil.rxSchedulerHelper())
    }


}
